### PR TITLE
Make satellite listen address fully customizable

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Attempts to solve #3505

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
@indualagarsamy this should cover your needs.

File: src/NServiceBus.Core/Features/FeatureConfigurationContext.cs [hunk 1]
```diff
@@ -41,12 +41,31 @@ internal FeatureConfigurationContext(ReadOnlySettings settings, IConfigureCompon
         /// <summary>
         /// Creates a new satellite processing pipeline.
         /// </summary>
+        /// <param name="name">Name of the satellite.</param>
+        /// <param name="transportAddress">Transport address to listen on.</param>
```
Comment (danielmarbach) [inline src/NServiceBus.Core/Features/FeatureConfigurationContext.cs]: Should we state here something like

> The autogenerated transport address to listen on.

and on the other method

> The manually assigned transport address to listen on.

Comment (SzymonPobiega) [inline src/NServiceBus.Core/Features/FeatureConfigurationContext.cs]: :+1:

**Comment by SimonCropp** [comment]
@SzymonPobiega  any related doco needed?

**Comment by andreasohlund** [comment]
Looks good

**Comment by SzymonPobiega** [comment]
@SimonCropp @indualagarsamy mentioned she's working on satellite doco and lack of this feature was blocking her. I'll sync up with her tomorrow to ask if the planned to cover this as well or I should do it.

**Comment by SzymonPobiega** [comment]
BTW, I think we should document the satellites "as-is" even if we are going to abandon them in favor of some other thing (e.g. raw endpoints) in vNext.

File: src/NServiceBus.Core/Features/FeatureConfigurationContext.cs [hunk 2]
```diff
@@ -41,12 +41,31 @@ internal FeatureConfigurationContext(ReadOnlySettings settings, IConfigureCompon
         /// <summary>
         /// Creates a new satellite processing pipeline.
         /// </summary>
+        /// <param name="name">Name of the satellite.</param>
+        /// <param name="transportAddress">The autogenerated transport address to listen on.</param>
```
Comment (indualagarsamy) [inline src/NServiceBus.Core/Features/FeatureConfigurationContext.cs]: Can we keep the same order as the parameters to make it less confusing?

Comment (SzymonPobiega) [inline src/NServiceBus.Core/Features/FeatureConfigurationContext.cs]: You mean move the `transport address` to the end of the second overload?

Comment (indualagarsamy) [inline src/NServiceBus.Core/Features/FeatureConfigurationContext.cs]: @szymonpobiega - yes. The parameter list in the API comments should be in the same order as that of the function to follow along easier.

**Pull request merged**